lib/pos_mempool.go

```diff
@@ -0,0 +1,391 @@
+package lib
+
+import (
+	"github.com/dgraph-io/badger/v3"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+type PosMempoolStatus int
+
+const (
+	PosMempoolStatusRunning PosMempoolStatus = iota
+	PosMempoolStatusNotRunning
+)
+
+type Mempool interface {
+	Start() error
+	Stop()
+	IsRunning() bool
+	ProcessMsgDeSoTxn(msg *MsgDeSoTxn) error
+	AddTransaction(txn *MempoolTx) error
+	RemoveTransaction(txn *MempoolTx) error
+	GetTransaction(txHash *BlockHash) *MempoolTx
+	GetTransactions() []*MempoolTx
+	GetIterator() MempoolIterator
+	UpdateLatestBlock(blockView *UtxoView, blockNode *BlockNode)
+	UpdateGlobalParams(globalParams *GlobalParamsEntry)
+}
+
+type MempoolIterator interface {
+	Next() bool
+	Value() (*MempoolTx, bool)
```

Comment:
Same comment as below. Should this return *MsgDeSoTxn to hide the MempoolTx struct for external callers? It should make it easier for us to refactor the mempool implementation into its own package later on.
```suggestion
	Value() (*MsgDeSoTxn, bool)
```